.github/workflows/pymeasure_CI.yml

```diff
@@ -64,26 +82,15 @@ jobs:
           activate-environment: pymeasure
           auto-update-conda: true 
           use-only-tar-bz2: true  # for caching to work properly
-      - name: Install CI dependencies
-        shell: bash -l {0}
-        run: conda install flake8
-      - name: Conda info
-        shell: bash -l {0}
-        run: conda info
-      - name: Conda list
-        shell: bash -l {0}
-        run: conda list
-      - name: Build pymeasure
-        shell: bash -l {0}
-        run: python setup.py sdist bdist_wheel
-      - name: Install pymeasure
-        shell: bash -l {0}
-        run: python -m pip install dist/*.whl
+      - name: Install dev requirements
+        run: conda env update -f requirements/pymeasure_dev.yml
```

Comment:
Note that this now always pulls sphinx and flake8 dependencies into every testing CI job, even though we don't use them at all in those jobs.

Reply:
I was thinking we can centralize on the cache, since those packages should also get wrapped up and then shared with the RTD test. I agree it is an extra install here, but I'm not sure how optimized we need to be.

Reply:
Good question what is more efficient - smaller/fewer cache or faster install. I also though in terms of complexity of moving parts, would be good to keep docs & linting separate from code testing, e.g. sphinx dependency hiccups or whatnot would not fail/error our pytest tests.

Reply:
Do you have any insight how we stand on using our CI quota?
 [AFAICT](https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions) we have 2000 "actions minutes" per month, where linux workers count 1x, Win 2x, Macos 10x. One CI run apparently takes ~15 min Linux time, 20 Macos, 24 Win.
If I am correct, and this could become a problem, from that point of view we should probably a) optimise for runtime and b) probably reduce our build matrix, e.g. only test with latest python on Win and Mac, and run tests on Linux wherever possible.

Reply:
The thing is, my gut says that we would have easily blown through our quota this month if the above numbers are correct, so something seems strange.

Reply:
Ahhhh, "free for public repositories", so maybe there is no quota. Still, considering how Gitlab just slashed *their* free minutes from 2000 to 400 (or so), I think it would be good to be "efficient" citizens. 
I'm not sure how much we learn anyhow from the additional Win and Mac builds at lower python versions (as long as we test with the latest) - Python compat problems should be caught in the Linux builds, anyway.

Reply:
I believe that is only on private repos. :) For the pymeasure Org it shows that we have not used any of the 2k actions minutes yet.